eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferTypes.scala

```diff
@@ -266,23 +248,26 @@ object OfferTypes {
      */
     def apply(offer: Offer, amount: MilliSatoshi, quantity: Long, features: Features[InvoiceFeature], payerKey: PrivateKey, chain: ByteVector32): InvoiceRequest = {
       require(offer.chains.contains(chain))
-      require(quantity == 1 || offer.quantityMin.nonEmpty || offer.quantityMax.nonEmpty)
-      val tlvs: Seq[InvoiceRequestTlv] = Seq(
-        Some(Chain(chain)),
-        Some(OfferId(offer.offerId)),
-        Some(Amount(amount)),
-        if (offer.quantityMin.nonEmpty || offer.quantityMax.nonEmpty) Some(Quantity(quantity)) else None,
-        Some(PayerKey(payerKey.publicKey)),
-        if (!features.isEmpty) Some(FeaturesTlv(features.unscoped())) else None,
-      ).flatten
-      val signature = signSchnorr(InvoiceRequest.signatureTag, rootHash(TlvStream(tlvs), OfferCodecs.invoiceRequestTlvCodec), payerKey)
-      InvoiceRequest(TlvStream(tlvs :+ Signature(signature)))
+      require(quantity == 1 || offer.quantityMax.nonEmpty)
+      val tlvs: Seq[InvoiceRequestTlv] = InvoiceRequestMetadata(randomBytes32()) +: (offer.records.records.toSeq ++ Seq(
```

Comment:
That's because I want to order the TLVs by tags and `InvoiceRequestMetadata` has tag 0 so it should go in front whereas the other invoice request TLVs are all higher than the offer TLVs.

Reply:
Is it important to order them by tag? We don't really care, tags are a serialization thing, we're at a higher layer here aren't we? Is it to help reviewers verify that all fields are provided?

Reply:
I think it makes it easier to follow and it allows us to to test for equality in tests.